spec/latest/core/index.html

```diff
@@ -1069,6 +1069,46 @@ <h4>Member assertions</h4>
     <p class="note">It's important to point out that the <code>subject</code>, <code>property</code>
       and <code>object</code> predicates are defined within the Hydra namespace and are not
       <a href="https://www.w3.org/TR/rdf-schema/"><code>rdf</code></a> terms.</p>
+
+    <p>It is also possible to use <i>memberAssertion</i> predicate on the API documentation level,
+      by attaching this predicate to subclasses of the <i>Collection</i>, like in the example below.
+      Clients would understand that all members of collections which are instances of <i>api:UserCollections</i>
+      would in fact have <i>rdf:type api:User</i>.</p>
+
+    <pre class="example nohighlight"
+         data-transform="updateExample"
+         title="Member assertion block describes strongly typed collection">
+      <!--
+      {
+        "@context": [
+          "http://www.w3.org/ns/hydra/context.jsonld",
+          { "api": "http://api.example.com/api/documentation#" }
+        ],
+        "@id": "http://api.example.com/api/documentation",
+        "@type": "ApiDocumentation",
+        "supportedClass": [api:User, api:UserCollection],
+        "api:UserCollection": {
+          "subClassOf": "Collection",
+          ****"memberAssertion": {
+            "property": "rdf:type",
+            "object": "api:User"
+          }****,
+        }
+      }
+      -->
+    </pre>
+
+    <p>It is worth to mention that a strongly typed collection instance can have its own
+      member assertions. In such scenario, both <i>API documentation</i> level and instance
+      level assertions should be combined as neither makes the other obsolete.</p>
+
+    <p>More complex scenario would involve a class hierarchy, in which each class can
+      carry additional member assertions compared to it's base class. To discover all
+      available member assertion blocks client SHOULD traverse whole class hierarchy
+      to gather all of the member assertions. In order to take such a burden from clients
+      it is strongly recommended to provide all member assertion blocks to be provided
+      on each class level including it's base class blocks so the client does not have
+      to perform this traversing behavior.</p>
```

Comment:
I don't think this makes the much sense. Even if an API duplicates all assertions on all levels of a class hierarchy, the client does not have a way of knowing that. Thus, they will have to traverse the hierarchy nonetheless.

I think I would change this paragraph and require only that a client inspects explicit `rdf:type` of a collection resource. In the example below the client would only discover the `hydra:manages` of `PersonCollection`

```turtle
# resource
</people> a api:PersonCollection, hydra:Collection .

# api docs
api:PersonCollection
  rdfs:subClassOf api:AgentCollection ;
  hydra:manages [ hydra:property rdf:type ; hydra:object foaf:Person ] ;
.

api:AgentCollection
  rdfs:subClassOf hydra:Collection ;
  hydra:manages [ hydra:property rdf:type ; hydra:object foaf:Agent ] ;
.
```

Technically, a client could perform RDFS reasoning but in practice I would suggest that the server did that and served all materialised RDF types instead, to relieve the clients from performing additional processing

Reply:
Hmm - I see your point. I've softened this required from _SHOULD_ to _MAY_. Strong recommendation for the server to provide all the details is still in force, but I do not see a way of forcing developers to do so. The client still can do the traversing though.